File: Day1/day1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

int get_fuel_requirement(int module_mass);
int calculate_fuel_fuel(int module_mass);

int main() {
    std::fstream input_file;
    std::vector <int> modules;

    // Populate modules with values from input.txt
    input_file.open("input.txt", std::ios::in);
    if (!input_file){
        std::cout << "No file found.\n";
    } else {
        std::string line;
        while (getline(input_file, line)){
            modules.push_back(std::stoi(line));
        }
    }
    // int fr = get_fuel_requirement(1969);
    // std::cout << fr << std::endl;
    int part_one_weight {0};
    for (int i: modules){
        part_one_weight += get_fuel_requirement(i);
    }
    std::cout << "Part one:\n";
    std::cout << part_one_weight << std::endl;

    int part_two_weight {0};
    for (int i: modules){
        part_two_weight += calculate_fuel_fuel(i);
    }
    std::cout << "Part two:\n";
    std::cout << part_two_weight << std::endl;
}


int get_fuel_requirement(int module_mass) {
    int fuel_required {0};
    fuel_required = module_mass/3 -2;
    return fuel_required;
}

// def calculate_fuel_fuel(module_mass):
//     if get_fuel_requirement(module_mass) <= 0:
//         return 0
//     else:
//         fuel_needed = get_fuel_requirement(module_mass)
//         return fuel_needed + calculate_fuel_fuel(fuel_needed)
int calculate_fuel_fuel(int module_mass){
    if (get_fuel_requirement(module_mass) <= 0){
        return 0;
    }
    else {
        int fuel_needed = get_fuel_requirement(module_mass);
        return fuel_needed + calculate_fuel_fuel(fuel_needed);
    }
}

File: Day1/day1.py
import os


def get_fuel_requirement(module_mass):
    module_mass = int(module_mass)
    fuel_required = module_mass//3 -2
    return fuel_required

def calculate_fuel_fuel(module_mass):
    if get_fuel_requirement(module_mass) <= 0:
        return 0
    else:
        fuel_needed = get_fuel_requirement(module_mass)
        return fuel_needed + calculate_fuel_fuel(fuel_needed)

if __name__=='__main__':
    data = []
    fileName = os.path.join(os.getcwd(), 'Day1/input.txt')
    with open(fileName, "r") as f:
        data = f.read().strip().split('\n')

    total_fuel_required = [get_fuel_requirement(module) for module in data]
    print(sum(total_fuel_required))

    total_fuel_required_with_fuel = [calculate_fuel_fuel(module) for module in data]
    print(sum(total_fuel_required_with_fuel))